Source/WebKit/NetworkProcess/Classifier/WebResourceLoadStatisticsStore.cpp

```diff
@@ -1327,11 +1327,13 @@ void WebResourceLoadStatisticsStore::callUpdatePrevalentDomainsToBlockCookiesFor
             m_networkSession->networkProcess().parentProcessConnection()->send(Messages::NetworkProcessProxy::SetDomainsWithUserInteraction(domainsWithUserInteractionQuirk), 0);
         }
 
-        HashMap<TopFrameDomain, SubResourceDomain> domainsWithStorageAccessQuirk;
+        HashMap<TopFrameDomain, Vector<SubResourceDomain>> domainsWithStorageAccessQuirk;
         for (auto& firstPartyDomain : domainsToBlock.domainsWithStorageAccess.keys()) {
-            auto requestingDomain = domainsToBlock.domainsWithStorageAccess.get(firstPartyDomain);
-            if (NetworkStorageSession::loginDomainMatchesRequestingDomain(firstPartyDomain, requestingDomain))
-                domainsWithStorageAccessQuirk.add(firstPartyDomain, requestingDomain);
+            auto requestingDomains = domainsToBlock.domainsWithStorageAccess.get(firstPartyDomain);
```

Comment:
```
for (auto& [firsrtPartyDomain, requestingDomains] : domainsToBlock.domainsWithStorageAccess) {
    for (auto& requestingDomain : requestingDomains)
}
```
to avoid the lookup